creepy/subprocess/pypen.py

```diff
@@ -108,6 +131,22 @@ def detach(self):
             del self._send, self._recv
         except AttributeError:
             pass
+        try:
+            os.remove(self._out_path)
+            os.remove(self._in_path)
+        except (OSError, FileNotFoundError, AttributeError):
+            pass
+
+
+def _make_fifo(path: str | None = None):
+    """Open FIFO at specified path or create and open a new one."""
+    if path is None:
+        path = os.path.join('/tmp', secrets.token_urlsafe(16))
+        os.mkfifo(path)
+    in_fd = os.open(path, os.O_RDONLY | os.O_NONBLOCK)
+    out_fd = os.open(path, os.O_WRONLY)
+    fcntl.fcntl(in_fd, fcntl.F_SETFL, os.O_RDONLY)
```

Comment:
Please comment such complex to understand lines, explain why do you use specific flags, etc…